src/entity.ts

```diff
@@ -300,8 +303,12 @@ export class Entity {
 					value = Entity.createOrUpdate(ChildEntity.meta, state, this._context);
 			}
 		}
-		else if (prop.isList && Array.isArray(state) && Array.isArray(currentValue))
+		else if (prop.isList && Array.isArray(state) && Array.isArray(currentValue)) {
 			currentValue.splice(0, currentValue.length, ...state.map(s => this.serializer.deserialize(this, s, prop, this._context)));
+
+			// Mark property pendingInit = false, since the setter is not called
+			Property$pendingInit(this, prop, false);
+		}
```

Comment:
Ensure that pending init is set to false when a property has initial state provided when the object is constructed